base/show.jl

```diff
@@ -2223,6 +2223,15 @@ function showarg(io::IO, r::ReinterpretArray{T}, toplevel) where {T}
     print(io, ')')
 end
 
+# printing iterators from Base.Iterators
+
+function show(io::IO, e::Iterators.Enumerate)
```

Comment:
Should this be three-arg show?

Reply:
IIUC 2-arg `show` is required to be parseable, but both the old and the new outputs satisfy this criterion, so probably better use the most compact form. See #34734.

Reply:
If we use three-arg `show`, then `print` will use the old printing. Is that the desired behavior?

Reply:
As Milan says, since this still parses it should be fine to use as 2 arg i think.